ding/policy/qtransformer.py

```diff
@@ -0,0 +1,498 @@
+from typing import List, Dict, Any, Tuple, Union
+import copy
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch.distributions import Normal, Independent
+from ema_pytorch import EMA
+
+from ding.torch_utils import Adam, to_device
+from ding.rl_utils import v_1step_td_data, v_1step_td_error, get_train_sample, \
+    qrdqn_nstep_td_data, qrdqn_nstep_td_error, get_nstep_return_data
+from ding.model import model_wrap
+from ding.utils import POLICY_REGISTRY
+from ding.utils.data import default_collate, default_decollate
+
+from .sac import SACPolicy
+from .qrdqn import QRDQNPolicy
+from .common_utils import default_preprocess_learn
+
+from pathlib import Path
+from functools import partial
+from contextlib import nullcontext
+from collections import namedtuple
+
+import torch
+import torch.nn.functional as F
+import torch.distributed as dist
+from torch import nn, einsum, Tensor
+from torch.nn import Module, ModuleList
+from torch.utils.data import Dataset, DataLoader
+
+from torchtyping import TensorType
+
+from einops import rearrange, repeat, pack, unpack
+from einops.layers.torch import Rearrange
+
+from beartype import beartype
```

Comment:
we will not use `beartype` to validate runtime types in the current version, thus remove it in this PR